cpp/CMakeLists.txt

```diff
@@ -65,9 +65,12 @@ set(RAFT_COMPILE_LIBRARY_DEFAULT OFF)
 if(BUILD_TESTS OR BUILD_PRIMS_BENCH)
   set(RAFT_COMPILE_LIBRARY_DEFAULT ON)
 endif()
-option(RAFT_COMPILE_LIBRARY "Enable building raft shared library instantiations"
+option(RAFT_COMPILE_LIBRARY "Enable building raft library instantiations"
        ${RAFT_COMPILE_LIBRARY_DEFAULT}
 )
+option(RAFT_COMPILE_DYNAMIC_ONLY "Only build the shared library and skip the
```

Comment:
I think that this is fine for now, but we should standardize our CMake strategies here a bit. raft is not the only RAPIDS library that people sometimes want static and at other times shared. I haven't fully thought through the ramifications of supporting both builds at once vs simply switching on `BUILD_SHARED_LIBS` (which is what cudf does IIRC). It gets extra confusing to think about the exports and whether you really need to build these as two separate packages or not. 

I don't want to block this PR on this though. Maybe open a build-planning issue for it (or remind me to do so tomorrow if you aren't sure what it should look like).

Reply:
Yeah thank you, I agree we shouldn't block this PR on this. I'd appreciate if you could open the issue, sounds like you have a more specific set of ideas about this than what I'd probably write.

Reply:
I opened https://github.com/rapidsai/build-planning/issues/134.